### PR TITLE
UN-2885 [FIX] Fix MinIO cleanup failure due to incorrect organization ID format in workflow execution context

### PR DESCRIPTION
**Pull request by muhammad-ali-e**
## What

- Fixed MinIO cleanup failures caused by organization ID format mismatch
- Changed `_get_organization_context()` method in `backend/workflow_manager/internal_views.py` to return correct organization ID format
- Single line change: `str(org.id)` → `str(org.organization_id)`

## Why

- Tools write files to MinIO paths using string organization IDs (e.g., `unstract/execution/org_HJqNgodUQsA99S3A/...`)
- Backend was returning numeric organization IDs (e.g., "2") in workflow execution context
- This caused cleanup operations to look in wrong paths (`unstract/execution/2/...`)
- Result: Intermittent `ToolOutputNotFoundException` errors and failed cleanup operations

## How

- Modified `backend/workflow_manager/internal_views.py:443`
- Changed from: `"organization_id": str(org.id)` (returns numeric "2")
- Changed to: `"organization_id": str(org.organization_id)` (returns "org_HJqNgodUQsA99S3A")
- This ensures MinIO paths match between file creation and cleanup operations

## Can this PR break any existing features. If yes, please list possible items. If no, please explain why. (PS: Admins do not merge the PR without this section filled)

**No, this PR will NOT break existing features. Here's why:**

1. **Workers are designed to handle string organization IDs**: Comprehensive code review shows all worker dataclasses expect `organization_id: str` and use defensive `str()` conversions
2. **No numeric operations on organization_id**: Grep search found zero instances of numeric operations or comparisons on organization_id in workers
3. **This is a bug fix, not a feature change**: The current behavior is incorrect and causes failures
4. **Path consistency**: This fix ensures paths are consistent with how the system actually stores files
5. **Single endpoint affected**: Only `WorkflowExecutionInternalViewSet.retrieve()` is modified, which is only used by workers
6. **Backward compatible**: String organization IDs work everywhere numeric IDs were used (but not vice versa)

## Database Migrations

- None required (no schema changes)

## Env Config

- None required (no environment variable changes)

## Relevant Docs

- Related investigation documented in issue UN-2885
- MinIO path format: `unstract/execution/{organization_id}/{workflow_id}/{execution_id}/`

## Related Issues or PRs

- Fixes UN-2885
- Related to MinIO cleanup failures and intermittent ToolOutputNotFoundException errors

## Dependencies Versions

- None changed

## Notes on Testing

- **Verification**: Check MinIO cleanup logs after deployment to confirm organization ID format is correct
- **Test scenario**: Execute workflows and verify cleanup operations use paths like `org_XXX` instead of numeric IDs
- **Expected behavior**: No more cleanup failures due to organization ID mismatch
- **Edge cases tested**: Workers handle string organization IDs correctly (confirmed via code review)

## Screenshots

N/A (Backend fix, no UI changes)

## Checklist

I have read and understood the [Contribution Guidelines](https://docs.unstract.com/unstract/contributing/unstract/).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: backend/workflow_manager/internal_views.py [hunk 1]
```diff
@@ -440,14 +440,18 @@ def _get_organization_context(self, execution: WorkflowExecution) -> dict:
             if execution.workflow and hasattr(execution.workflow, "organization"):
                 org = execution.workflow.organization
                 return {
-                    "organization_id": str(org.id),
-                    "organization_name": org.display_name,
+                    "organization_id": str(
+                        org.organization_id
+                    ),  # organization identifier
+                    "organization_uuid": str(org.id),  # organization uuid
+                    "organization_name": org.display_name,  # organization name
```
[the remark this answers is not in the document]
Comment (muhammad-ali-e) [reply backend/workflow_manager/internal_views.py:447]: In the minio path we rquired the organization_id as string based like org_*.  not id.

**Review by johnyrahul — APPROVED** [review]
